TreasurePool new addon

KenshiDRK opened the pull request:
Replacement for the in game treasure pool with aditional information.

Commit:
Initial commit

Commit:
Create ReadMe.md

Commit:
Update addons.xml

Commit:
Update TreasurePool.lua

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+end)
+
+windower.register_event('prerender', function()
+    local get_item = windower.ffxi.get_items()
```
Since you only need treasure from this you can do:

``` lua
    local treasure = T(windower.ffxi.get_items().treasure)
```

Then use that directly in the following code.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+        return
+    end
+    for i,v in pairs(get_item.treasure) do
+        for i = 0, 9 do
```
You only need one of these two loops.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+    local get_item = windower.ffxi.get_items()
+    local remove = S{}
+    local info = {}
+    if zoning_bool or (not get_item.treasure[0] and not get_item.treasure[1] and not get_item.treasure[2] and not get_item.treasure[3] and not get_item.treasure[4] and not get_item.treasure[5] and not get_item.treasure[6] and not get_item.treasure[7] and not get_item.treasure[8] and not get_item.treasure[9]) then
```
Quicker:

``` lua
    if zoning_bool or T(get_item.treasure):empty() then
```

Or, if you implemented the above advice:

``` lua
    if zoning_bool or treasure:empty() then
```

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+                            or diff > 180 and
+                                '\\cs(0,255,0)' .. item[i] .. ' → ' .. timer[i]
+                            or
+                                '\\cs(255,128,0)' .. item[i] .. ' → ' .. timer[i]) .. '\\cr'
```
This can be done prettier, but I'll need to explain a few other things first. Hit me up on IRC about it at some point during the weekend.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+    
+    if id == 0x0D3 then
+ 
+    local lotpacket = packets.parse('incoming', data)
```
Forgot to indent this.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+    local packet = packets.parse('incoming', data)
+    local time_check = packet.Timestamp + 300
+    local diff = os.difftime(time_check, os.time())
+        -- Ignore gil drop
```
This can be moved before the `time_check` declaration, right after `packet`.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,171 @@
+
+    if id == 0x0D2 then
+ 
+    local packet = packets.parse('incoming', data)
```
The indentation here is off.

Commit:
Fixes

I think thats all except for the color code, hit me up if you want me to change it.

Commit:
copy paste is no good and last line again >.<

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,160 @@
+--[[Copyright © 2015, Kenshi
```
Yes, it has been _that_ long since you requested this :D Please change this to 2016.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,160 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of BCTimer nor the
```
Also change the name of the addon.

z16 commented on addons/TreasurePool/TreasurePool.lua:
```diff
@@ -0,0 +1,160 @@
+    local info = {}
+    if zoning_bool or treasure:empty() then
+        remove:add('index0') remove:add('index1') remove:add('index2') remove:add('index3') remove:add('index4') remove:add('index5') remove:add('index6') remove:add('index7') remove:add('index8') remove:add('index9')
+        remove:add('lotting0') remove:add('lotting1') remove:add('lotting2') remove:add('lotting3') remove:add('lotting4') remove:add('lotting5') remove:add('lotting6') remove:add('lotting7') remove:add('lotting8') remove:add('lotting9')
```
Please disregard what I wrote here earlier, this will not have the same effect as what you were doing. I was thinking you were trying to do something else.

KenshiDRK replied:
Indeed those lines are redundant and I erased them, the code also changed a little to try to avoid some weird arithmetic on nil value error. I'll update it in a min.

z16 replied:
Glad you figured out what I meant, despite deleting more of my original text than I intended :)

Commit:
Update

Update

alphaONE2 commented:
@z16, @KenshiDRK: What's the status of this Addon? Can we go ahead and merge this?

z16 commented:
Forgot all about it! Looks ready in my eyes, so I'll go ahead and merge it.
